### PR TITLE
13865 safari block cookies breaks javascript scripts

### PR DESCRIPTION
**Pull request by raulvOnestic91**
Create a popup to warn that the cookies are blocked in the browser.

### Description (*)

I fixed this PR creating a js that checks if the navigator has the cookies blocked and warn with a popup that the behavior could be wrong.

### Fixed Issues (if relevant)

1. #13865: Safari "Block all cookies" setting breaks JavaScript scripts

### Manual testing scenarios (*)

1. Open Safari browser and go to Preferences
2. In Privacy tab enable Block all cookies setting.
3. Open any Magento shop.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [] All new or changed code is covered with unit/integration tests (if applicable)
 - [] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by raulvOnestic91** [comment]
hello @DrewML @melnikovi I not able to see what is wrong in the Static Tests, the details page is not charging. Could you tell me what is the error please?

**Comment by Karlasa** [comment]
@magento run all tests

File: app/code/Magento/Theme/view/frontend/web/js/cookie-status.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    $.widget('mage.cookieStatus', {
+        /**
+         * Init object
+         * @private
+         */
+        _init: function () {
+
+            if(!navigator.cookieEnabled) {
+                const options = {
```
Comment (krzksz) [inline app/code/Magento/Theme/view/frontend/web/js/cookie-status.js]: Could you please move those options to widget's `options` field so people are able to customize them without overwriting the whole method?

**Comment by krzksz** [comment]
@magento run all tests

File: app/code/Magento/Theme/view/frontend/web/js/cookie-status.js (hunk 1 above)
Comment (krzksz) [inline app/code/Magento/Theme/view/frontend/web/js/cookie-status.js]: Can we move modal settings to widget's `options` field so people are able to customize them without overwriting the whole method?

File: app/code/Magento/Theme/view/frontend/web/js/cookie-status.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+define([
+    'jquery',
+    'mage/translate',
```
Comment (krzksz) [inline app/code/Magento/Theme/view/frontend/web/js/cookie-status.js]: Just move this dependency as the last one, so you don't need to introduce `$tr` which is not used anyway.

**Review by krzksz — CHANGES_REQUESTED** [review]

File: app/code/Magento/Theme/view/frontend/web/js/cookie-status.js (hunk 1 above)
Comment (raulvOnestic91) [reply app/code/Magento/Theme/view/frontend/web/js/cookie-status.js]: Done

File: app/code/Magento/Theme/view/frontend/web/js/cookie-status.js (hunk 2 above)
Comment (raulvOnestic91) [reply app/code/Magento/Theme/view/frontend/web/js/cookie-status.js]: Done

**Review by krzksz — APPROVED** [review]

**Review by krzksz — CHANGES_REQUESTED** [review]
Please take a look at failing static tests.

**Comment by magento-engcom-team** [comment]
Hi @krzksz, thank you for the review.
[ENGCOM-6378](https://jira.corp.magento.com/browse/ENGCOM-6378) has been created to process this Pull Request
:eight_spoked_asterisk: @krzksz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by krzksz** [comment]
Hey @magento-engcom-team, I am not able to add a tests label as I am not sure if it is possible to simulate disabled cookies in MFTF.

**Comment by VladimirZaets** [comment]
Hi @raulvOnestic91, @krzksz. In this case, first of all, I think we should create a separate class that will wrap the native navigator class. 
It will give a few benefits:
1) We will control the functionality of Navigator and will have the ability to add/change functionality in just one place.
2) It will five ability to cover functionality at least by the unit test

**Comment by engcom-Golf** [comment]
I will take care of test coverage

File: dev/tests/js/jasmine/tests/app/code/Magento/Theme/view/frontend/web/js/cookie-status.test.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            $(document.body).append(htmlContainer);
+            widget = new Cookie();
+            navigator = window.navigator;
+        });
+
+        afterEach(function () {
+            window.navigator = navigator;
+        });
+
+        it('verify cookie-status initialization', function () {
```
Comment (krzksz) [inline dev/tests/js/jasmine/tests/app/code/Magento/Theme/view/frontend/web/js/cookie-status.test.js]: I would change that to `it('shows the modal when cookies are not supported'` and then add a test that modal is not added when cookies are supported.

File: dev/tests/js/jasmine/tests/app/code/Magento/Theme/view/frontend/web/js/cookie-status.test.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            expect($.fn.cookieStatus).toBeDefined();
+            window.navigator = {
+                cookieEnabled: false
+            };
+            widget._init();
+            expect($('.cookie-status').length).toBe(1);
+            expect($(document.body).html()).toContain('<aside role="dialog" class="modal-popup');
```
Comment (krzksz) [inline dev/tests/js/jasmine/tests/app/code/Magento/Theme/view/frontend/web/js/cookie-status.test.js:46]: You can check for `$('.modal-popup').length` or `$('aside.modal-popup').length` or `$('aside[role="dialog"]').length` depending on how strict you need to be.

File: dev/tests/js/jasmine/tests/app/code/Magento/Theme/view/frontend/web/js/cookie-status.test.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            widget = new Cookie();
+            navigator = window.navigator;
+        });
+
+        afterEach(function () {
+            window.navigator = navigator;
+        });
+
+        it('verify cookie-status initialization', function () {
+            expect($.fn.cookieStatus).toBeDefined();
```
Comment (krzksz) [inline dev/tests/js/jasmine/tests/app/code/Magento/Theme/view/frontend/web/js/cookie-status.test.js:30]: I would extract that check to a separate `it('defines cookieStatus widget'`

**Review by krzksz — CHANGES_REQUESTED** [review]

File: dev/tests/js/jasmine/tests/app/code/Magento/Theme/view/frontend/web/js/cookie-status.test.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+        afterEach(function () {
+            window.navigator = navigator;
+        });
+
+        it('verify initialization cookieStatus widget', function () {
+            expect($.fn.cookieStatus).toBeDefined();
+        });
+
+        it('verify that modal does not shows when cookies are supported', function () {
```
Comment (krzksz) [inline dev/tests/js/jasmine/tests/app/code/Magento/Theme/view/frontend/web/js/cookie-status.test.js]: Please change to `does not show a modal when cookies are supported`.

File: dev/tests/js/jasmine/tests/app/code/Magento/Theme/view/frontend/web/js/cookie-status.test.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+            $(document.body).append(htmlContainer);
+            widget = new Cookie();
+            navigator = window.navigator;
+        });
+
+        afterEach(function () {
+            window.navigator = navigator;
+        });
+
+        it('verify initialization cookieStatus widget', function () {
```
Comment (krzksz) [inline dev/tests/js/jasmine/tests/app/code/Magento/Theme/view/frontend/web/js/cookie-status.test.js]: Please change to `defines cookieStatus widget`.

**Review by krzksz — CHANGES_REQUESTED** [review]

**Review by krzksz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @krzksz, thank you for the review.
[ENGCOM-6378](https://jira.corp.magento.com/browse/ENGCOM-6378) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Golf** [comment]
Failed functional test not related to the changes in this PR, test fails on 2.4-develop branch.

**Pull request merged**